src/storage/src/hummock/shared_buffer/shared_buffer_batch.rs

```diff
@@ -724,8 +730,8 @@ impl<D: HummockIteratorDirection> HummockIterator for SharedBufferBatchIterator<
     }
 
     fn key(&self) -> FullKey<&[u8]> {
-        let (key, (epoch, _)) = self.current_item();
-        FullKey::new(self.table_id, TableKey(key), *epoch)
+        let (key, (epoch_with_gap, _)) = self.current_item();
+        FullKey::new(self.table_id, TableKey(key), epoch_with_gap.as_u64())
```

Comment:
For this key, we'd better have a `new_with_gap` to accept the EpochWithGap struct instead of a u64. The original FullKey::new should be consistent to use pure_epoch as input.